pkg/resource/deploy/step_generator.go

```diff
@@ -298,11 +298,10 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, res
 			if !isAliased {
 				urn = s.URN()
 			}
-			resourcePlan, ok := sg.deployment.newPlans.get(urn)
-			if !ok {
-				return nil, result.Errorf("Expected a new resource plan for %v", urn)
+			if resourcePlan, ok := sg.deployment.newPlans.get(urn); ok {
+				// If the resource is in the plan, add the operation to the plan.
+				resourcePlan.Ops = append(resourcePlan.Ops, s.Op())
```

Comment:
to gate on isTargeted below means that urns are not added to this. This is caught above by checking that unspecified resources are sames.